### PR TITLE
universe v3 equivalent of ListHandler

### PR DESCRIPTION
**Pull request by tamarrow**
#371

### CODE REVIEW

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Universe.scala [hunk 1]
```diff
@@ -22,6 +22,71 @@ object Universe {
...
+      universe.v3.model.V3PackagingVersion] = {
+
+    Bijection.build { (_: universe.v2.model.PackagingVersion) =>
+      universe.v3.model.V3PackagingVersion.instance
+    } { (x: universe.v3.model.V3PackagingVersion) =>
+      universe.v2.model.PackagingVersion(x.v)
+    }
+  }
+
+  implicit val v3PackagingVersionToV2PackagingVersion: Injection[
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Universe.scala]: The name of this and the one above should be swapped.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Response.scala [hunk 2]
```diff
@@ -38,5 +38,20 @@ object Response {
...
+    }
+
+  def v2ListResponseToV1ListResponse(x: rpc.v2.model.ListResponse): rpc.v1.model.ListResponse = {
+    rpc.v1.model.ListResponse(
+      packages = x.packages.map { y =>
+        rpc.v1.model.Installation(
+          appId = y.appId,
+          packageInformation = rpc.v1.model.InstalledPackageInformation(
+              packageDefinition = y.packageInformation.as[universe.v2.model.PackageDetails],
+              resourceDefinition = y.packageInformation.as[Option[universe.v2.model.Resource]]
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Response.scala]: This usage of `as` seems to go against the intended use of the Bijection library. Here, we're taking a `universe.v3.model.PackageDefinition` and "converting" it to a `universe.v2.model.Resource`. But there's not a valid bijection or injection between the two types. Instead, we probably should do something like this:

```scala
resourceDefinition = y.packageInformation.resource.as[Option[universe.v2.model.Resource]]
```

That would use the `Bijection/Injection` that already exists for `Resource`.

Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Response.scala]: Ah, part of the problem is that `packageInformation.resource` doesn't exist, we'd have to pattern match. That's still better handled outside of `Bijection/Injection` IMO.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala [hunk 3]
```diff
@@ -14,3 +17,16 @@ trait CosmosRepository extends PackageCollection {
...
+// TODO (version): Rename to CosmosRepository
+/** A repository of packages that can be installed on DCOS. */
+trait V3CosmosRepository extends PackageCollection {
+
+  def repository: PackageRepository
+
+  def getPackageByReleaseVersion(
+    packageName: String,
+    releaseVersion: ReleaseVersion
+  ): Future[(V3Package, Uri)]
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala]: We don't need to include the `Uri`, right?

**Comment by cruhland** [comment]
@tamarrow just a few comments, good to merge once they are addressed. Don't worry too much about the `Bijection` stuff, we'll probably clean some of that up in a separate PR.

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala (hunk 3 above)
Comment (tamarrow) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala]: removed

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Universe.scala (hunk 1 above)
Comment (tamarrow) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Universe.scala]: done

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Response.scala (hunk 2 above)
Comment (tamarrow) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/converter/Response.scala]: updated

**Comment by tamarrow** [comment]
thanks @cruhland

**Pull request merged**